### DISCUSSION TITLE
Feature: What about hints in IDE?

### DISCUSSION THREAD

**Discussion by alex-kovalchuk** [ngneat/transloco#469] [Ideas]
Hello. It would be very convenient to find the desired translation in .json files by ctrl + click on the key in selectTranslate () or in templates.

I'm using Phpstorm and at the moment I have this problem:

<img width="797" alt="Снимок экрана 2021-04-13 в 11 00 35" src="https://user-images.githubusercontent.com/14261969/114519169-cc0e4e00-9c48-11eb-98cc-de2ff118f05c.png">

Should I expect any plugins for Jetbrains IDEs or other?
Maybe there are some other approaches?

Thanks!!!

**Comment by mattmutt** [comment]
In an unrelated project for work, I've solved this concern with straight plain old TypeScript bindings. Here is what the developer sees in Webstorm, and VS Code is rather similar. No plugins, no fuss trying to guess obscure keys. Bringing typesafe i18n makes it easier for developers.

![angular-template-language-service-organized-keys-hint](https://user-images.githubusercontent.com/8688570/115772426-f2786b80-a363-11eb-95f5-a719a0c4e745.png)

**Comment by NiklasPor** [comment]
FYI [i18n-ally](https://github.com/lokalise/i18n-ally) is a VSCode extension and supports this (at least in templates).

**Comment by MickL** [comment]
This is definitely possible with TypeScript, I have seen this within: https://github.com/mozilla/node-convict/tree/master/packages/convict but I dont remember how this was called
